### ISSUE TITLE
Team Platform - Sprint 1.30.0 2/2

### ISSUE THREAD

**Issue by yakkomajuri** [PostHog/posthog#6183]
Team Platform

## Retro

- Meeting in person + fighting fires helped integrate the team more
- Unable to ship a lot of stuff that we planned to as we had to prioritize handling the ingestion issues 
- Team members who were in person were more inaccessible to Tiina
    - Ensuring we plan appropriately for coordinating sync and async work (particularly with in-person meetings)
- @tiina303 was able to ship iteratively and prioritize to keep focus on the objective (Java library)

## Hang over items from previous sprint

- Reprioritized: Splitting CH out of Helm Chart
- Carried over: `properties_last_updated_at`
- Continue: Java library
   - Docs (+ instructions for other JVM languages)
   - Testing
   - Feature flags??

### High priority

- @tiina303: https://github.com/PostHog/posthog/issues/5977
- @fuziontech: Kafka ingestion tests
- @fuziontech: Events table migration to unblock upgrades
- @yakkomajuri: Dead letter queue
- @yakkomajuri: `properties_last_updated_at` (carried over)
- @guidoiaquinti: Completing onboarding
- @guidoiaquinti: #6095 

### Low priority / side quests

https://github.com/orgs/PostHog/projects/10

**Comment by macobo** [comment]
Note from core analytics side: Being behind on clickhouse versions is becoming more painful by the week (also a retro point from this sprint). 

This is blocked by https://github.com/PostHog/posthog/issues/5684 + making sure the chart users can update.

**Comment by fuziontech** [comment]
From Karl on Slack on examples of pain points:
- I spent 3 hours fighting with clickhouse-copier yesterday before giving up and using tsvs. Fixed in a recent version
- Recent versions have multi-if statements which would speed up some queries
- Similar - pushing down LIMIT clauses on distributed queries
- There's new and better tooling around materialized columns and backfilling data, could kill some cruft
- Better EXPLAIN output
- M1 support

**Comment by neilkakkar** [comment]
Another important one: Fixes to how window functions work are in the newer version, so if we upgrade, we can 

(1) Ensure our queries are more accurate.
(2) Get rid of certain workarounds (without which some results were very very wrong, like negative times to convert))
